src/configure/configure.ts

```diff
@@ -116,54 +116,75 @@ class Orchestrator {
         }
     }
 
+    private async getAzureResource(targetType: TargetResourceType) {
+        var azureResourceSelector = ResourceSelectorFactory.getAzureResourceSelector(targetType);
+        var resource = null;
+        try {
+            resource = await azureResourceSelector.getAzureResource(this.inputs);
+            return resource;
+        }
+        catch (err) {
+            if (err.message === Messages.setupAlreadyConfigured) {
+                this.continueOrchestration = false;
+                return;
+            }
+            else {
+                throw err;
+            }
+        }
+    }
+
+    private async selectTemplate(resource: GenericResource) {
+        switch (resource.type) {
+            case TargetResourceType.AKS:
+                let templateParameterHelper = new TemplateParameterHelper();
+                let template = this.inputs.potentialTemplates[0] as LocalPipelineTemplate;
+                this.inputs.pipelineConfiguration.template = this.inputs.potentialTemplates[0];
+                await templateParameterHelper.setParameters(template.parameters, this.inputs);
+                break;
+
+            case TargetResourceType.WebApp:
+                var shortlistedTemplates = [];
+                shortlistedTemplates = this.inputs.potentialTemplates.filter((template) => template.targetKind === resource.kind);
+                if (shortlistedTemplates.length > 1) {
+                    this.inputs.pipelineConfiguration.template = shortlistedTemplates.find((template) => template.templateType === TemplateType.REMOTE);
+                }
+                else {
+                    this.inputs.pipelineConfiguration.template = shortlistedTemplates[0];
+                }
+                break;
+
+            default:
+                throw new Error("Resource not supported");
```

Comment:
I think we define all error messages in a different file

Reply:
done